pkg/generator/ingress_translator.go

```diff
@@ -259,12 +265,38 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 			}
 		}
 	}
+	listener := ""
+	port := uint32(0)
+
+	if config.FromContextOrDefaults(ctx).Kourier.TrafficIsolation == pkgconfig.IsolationIngressPort {
+		ns, err := translator.namespaceGetter(ingress.Namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		if ns.Labels != nil {
+			listener = ns.Labels[pkgconfig.TenantNameLabelKey]
+			if value, ok := ns.Annotations[pkgconfig.ListenerPortAnnotationKey]; ok {
+				p, err := strconv.ParseInt(value, 10, 32)
+				if err != nil {
+					return nil, err
+				}
+				port = uint32(p)
+
+				logger.Infof("mapping ingress %s/%s to port %d", ingress.Namespace, ingress.Name, port)
+			}
+		}
+
+		// REVISIT: When neither labels/annotations if found then default to the default behavior (no isolation)
+	}
 
 	return &translatedIngress{
 		name: types.NamespacedName{
 			Namespace: ingress.Namespace,
 			Name:      ingress.Name,
 		},
+		listener:                listener,
+		port:                    port,
 		sniMatches:              sniMatches,
 		clusters:                clusters,
 		externalVirtualHosts:    externalHosts,
```

Comment:
Maybe not for this PR, but it feels like this method might work better to construct the return value at the top (with default values for all the calculated fields), and then fill in the struct as it moves through the function. This would avoid needing to (for example) pre-fill `listener`, `port`, `sniMatches`, etc.